### PR TITLE
[JENKINS-70874] Fix 'Text file busy' issue

### PR DESCRIPTION
**Pull request by ckullabosch**
https://issues.jenkins.io/browse/JENKINS-70874

### Testing done

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by ckullabosch** [comment]
@jglick My understanding is that this not a JDK level problem. It's an inherent problem, when forking and writing executable files concurrently (independent of the language used). The issue in https://bugs.openjdk.org/browse/JDK-8068370 is probably the same root cause (but not fixable inside the JVM). You are right that lib-durable-task needs a fix as well. Please see here for more background information: https://issues.jenkins.io/browse/JENKINS-48258?focusedId=336050&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-336050

File: src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java [hunk 1]
```diff
@@ -229,27 +231,32 @@ private List<String> scriptLauncherCmd(ShellController c, FilePath ws, @CheckFor
         if (os == OsType.WINDOWS) { // JENKINS-40255
             scriptPath = scriptPath.replace("\\", "/"); // cygwin sh understands mixed path  (ie : "c:/jenkins/workspace/script.sh" )
         }
+        String scriptPathCopy = scriptPath + ".copy"; // copy file to protect against "Text file busy", see JENKINS-70874
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java:234]: Ought to be covered by https://github.com/jenkinsci/durable-task-plugin/blob/8ea90b6f96daddccc18e4622eb289675247e213c/src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java#L487 I think.

**Review by jglick — APPROVED** [review]
Patch itself seems reasonable. Will leave open for a bit to see if anyone else cares to comment on the approach.

**Comment by jglick** [comment]
(Can I assume that the problem was reproducible locally without this patch, and corrected when running with this patch? If you are not familiar with Jenkins plugin development generally, the easiest way is to use `mvn hpi:run`.)

**Comment by ckullabosch** [comment]
Yes, I did run tests for multiple hours and have not seen the issue again after applying the fix. Nevertheless it's happening very rare and is hard to reproduce.

**Review by car-roll — APPROVED** [review]

**Comment by ckullabosch** [comment]
The git-client plugin needs also a fix for the "Text file busy" issue, please see https://github.com/jenkinsci/git-client-plugin/pull/1103 and https://issues.jenkins.io/browse/JENKINS-72667.

**Pull request merged**